components/batch-system/src/batch.rs

```diff
@@ -509,6 +554,21 @@ where
         &self.router
     }
 
+    pub fn build_pool_state<H: HandlerBuilder<N, C>>(
+        &mut self,
+        handler_builder: H,
+    ) -> Option<PoolState<N, C, H>> {
```

Comment:
Perhaps `build_pool_state` should not return `None`?

Reply:
Fixed, thanks!